src/ISDeviceCal.h

```diff
@@ -0,0 +1,243 @@
+    void resize(int rows, int cols)
+    {
+        m = rows;
+        n = cols;
+        mat.resize(m*n);
+        zero();
+    }
+
+    float *matrix()
+    { 
+        return &mat[0]; 
+    }
+
+    void zero()
+    {
+        for (size_t i=0; i<mat.size(); i++)
+            mat[i] = 0;
+    }
+
+    void debug()
+    {
+        for (size_t i=0; i<mat.size(); i++)
+            mat[i] = i+1;
+    }
+
+    void fromString(std::string str)
+    {
+        StringToMatrix(str, &mat[0], m, n);
+    }
+
+    std::string toString()
+    {
+        return MatrixToString(&mat[0], m, n);
+    }
+};
+
+struct sCalData 
+{
+    sMatrix                 Y;          // truth data
+    sMatrix                 Ahat;       // matrix we're solving for
+    sMatrix                 Xhat;       // sampled/uncalibrated data
+    int                     len;
+
+    sCalData()
+    {
+
+    }
+
+    sCalData(int size)
+    {
+        len = size;
+        Y.resize(3, size);
+        Ahat.resize(3, 4);
+        Xhat.resize(4, size);
+    }
+
+    void resize(int size)
+    {
+        len = size;
+        Y.resize(3, size);
+        Ahat.resize(3, 4);
+        Xhat.resize(4, size);
+    }
+
+    void zero()
+    {
+        Y.zero();
+        Ahat.zero();
+        Xhat.zero();
+    }
+
+    void debug()
+    {
+        Y.debug();
+        Ahat.debug();
+        Xhat.debug();
+    }
+};
+
+// struct sMpuCalDataPtr
+// {
+//     sCalData *gyr;
+//     sCalData *acc; 
+//     sCalData *mag;
+// };
+
+struct sOrthoCal
+{
+    sCalData gyr[NUM_IMU_DEVICES];
+    sCalData acc[NUM_IMU_DEVICES];
+    sCalData mag[NUM_MAG_DEVICES];
+
+    sOrthoCal()
+    {
+        for (int d = 0; d < NUM_IMU_DEVICES; d++)
+        {
+            gyr[d].resize(NUM_PQR_SAMPLES);
+            gyr[d].zero();
+            acc[d].resize(NUM_POSES);
+            acc[d].zero();
+        }
+        for (int d = 0; d < NUM_MAG_DEVICES; d++)
+        {
+            mag[d].resize(NUM_HDG_SAMPLES);
+            mag[d].zero();
+        }
+    }
+};
+
+/**
+ * @brief Device calibration class for loading and saving calibration data
+ * 
+ * This class provides functionality to load and save device calibration data
+ * from/to JSON files, including sensor calibration info, temperature compensation,
+ * and motion calibration (orthonormalization).
+ */
+class ISDeviceCal
+{
+public:
+    /**
+     * @brief Load calibration data from a JSON file
+     * 
+     * @param filePath Path to the JSON file to load
+     * @param ocal Pointer to orthonormalization calibration data (can be NULL)
+     * @param info Pointer to sensor calibration info (can be NULL)
+     * @param tcal Pointer to temperature compensation calibration (can be NULL)
+     * @param mcal Pointer to motion calibration (can be NULL)
+     * @param pose Pointer to current pose value (can be NULL)
+     * @return true if calibration was successfully loaded, false otherwise
+     */
+    static bool loadCalibrationFromJSON(const std::string& filePath, 
```

Comment:
loadCalibrationFromJSON() is ambiguous. We should distinguish between loading from File, and loading from a JSON string, and loading from a JSON object.  These should be a nested series of calls....

LoadFromFile() :: JsonStringFromFile() -> jsonObjFromString() -> CalFromJsonObj()

Also, for Austin's work on the calibration database, we won't be using files directly (of course, thats an option, but we'll handle a JSON string directly from the database/HTTP request).